geode-core/src/main/java/org/apache/geode/internal/cache/wan/serial/SerialGatewaySenderQueue.java

```diff
@@ -316,71 +304,66 @@ public AsyncEvent take() throws CacheException {
    * have peeked. If the entry was not peeked, this method will silently return.
    */
   @Override
-  public void remove() throws CacheException {
-    lock.writeLock().lock();
+  public synchronized void remove() throws CacheException {
+    if (peekedIds.isEmpty()) {
+      return;
+    }
+    Long key = peekedIds.remove();
+    boolean isExtraPeeked = extraPeekedIds.remove(key);
     try {
-      if (peekedIds.isEmpty()) {
-        return;
+      // Increment the head key
+      if (!isExtraPeeked) {
+        updateHeadKey(key.longValue());
       }
-      Long key = peekedIds.remove();
-      boolean isExtraPeeked = extraPeekedIds.remove(key);
-      try {
-        // Increment the head key
-        if (!isExtraPeeked) {
-          updateHeadKey(key.longValue());
-        }
-        removeIndex(key);
-        // Remove the entry at that key with a callback arg signifying it is
-        // a WAN queue so that AbstractRegionEntry.destroy can get the value
-        // even if it has been evicted to disk. In the normal case, the
-        // AbstractRegionEntry.destroy only gets the value in the VM.
-        this.region.localDestroy(key, WAN_QUEUE_TOKEN);
-        this.stats.decQueueSize();
-
-      } catch (EntryNotFoundException ok) {
-        // this is acceptable because the conflation can remove entries
-        // out from underneath us.
-        if (logger.isDebugEnabled()) {
-          logger.debug(
-              "{}: Did not destroy entry at {} it was not there. It should have been removed by conflation.",
-              this, key);
-        }
+      removeIndex(key);
+      // Remove the entry at that key with a callback arg signifying it is
+      // a WAN queue so that AbstractRegionEntry.destroy can get the value
+      // even if it has been evicted to disk. In the normal case, the
+      // AbstractRegionEntry.destroy only gets the value in the VM.
+      this.region.localDestroy(key, WAN_QUEUE_TOKEN);
+      this.stats.decQueueSize();
+
+    } catch (EntryNotFoundException ok) {
+      // this is acceptable because the conflation can remove entries
+      // out from underneath us.
+      if (logger.isDebugEnabled()) {
+        logger.debug(
+            "{}: Did not destroy entry at {} it was not there. It should have been removed by conflation.",
+            this, key);
       }
+    }
 
-      boolean wasEmpty = this.lastDispatchedKey == this.lastDestroyedKey;
-      if (!isExtraPeeked) {
+    boolean wasEmpty = this.lastDispatchedKey == this.lastDestroyedKey;
+    if (!isExtraPeeked) {
+      this.lastDispatchedKey = key;
```

Comment:
I was going to ask why this was re-ordered, but its just the diff making it look that way...